### DISCUSSION TITLE
자신만의 테스트 철학이 있는지?

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#24] [Test]
.

**Comment by gyungchan-jo** [comment]
먼저 코드의 각각의 기능이 의도한 대로 작동하는지를 검증하는 단위 테스트가 중요하다 생각합니다. 이를 통해 작은 단위에서 버그를 발견하고 수정할 수 있으며, 코드의 안정성과 신뢰성을 향상시킬 수 있습니다.

또한 기능을 구현하기 전 테스트를 먼저 작성하는 것이 좋다고 생각합니다. 이를 테스트 주도 개발(TDD)라 하는데, 테스트 주도 개발은 소프트웨어를 개발할 때 테스트 케이스를 먼저 작성하고, 그에 맞는 코드를 작성하는 방법론입니다. 이를 통해 코드의 품질을 유지하고 버그를 최소화할 수 있습니다.

**Comment by le2sky** [comment]
(테스트의 필요성)
만약 테스트 코드가 없다면, 경험과 감에 의존하는 현상이 발생할 수 있습니다.
또한, 문제에 대한 피드백이 늦게 돌아올 수 있습니다.
이는 곧 유지보수하기 어려운 코드를 만들고, 소프트웨어의 신뢰성을 하락시킬 수 있습니다.

(올바른 테스트의 필요성)
개발자는 테스트를 통해서, 빠른 피드백과 자동화, 안정감을 얻을 수 있습니다.
다만, 테스트를 작성할 때, 느끼는 안정감이 잘못된 안정감인지 의식해야합니다.
잘못된 안정감으로 인해, 예기치 못한 상황이 초래될 수 있습니다.

(테스트는 문서가 될 수 있다)
테스트 코드는 프로덕션 기능을 설명하는 문서가 될 수 있습니다.
다양한 테스트 케이스를 통해 프로덕션 코드를 이해하는 시각과 관점을 보완할 수 있습니다.
어느 한 사람이 과거에 경험했던 고민의 결과물을 팀 차원으로 승격시켜서, 모두의 자산으로 공유할 수 있습니다.

(타협하지 않는 마음)
테스트를 작성하기 귀찮은 마음과 시간의 압박이 있을 수 있습니다.
하지만, 현재 시간을 더욱 투자하여 테스트 코드를 작성하면, 나중에 더 많은 시간을 아낄 수 있습니다.

**Comment by GIVEN53** [comment]
[좋은 테스트 코드란](https://github.com/2024-woowacourse-study/level-interview/discussions/42)과 겹치는 느낌
